packages/hoppscotch-sh-admin/src/pages/users/index.vue

```diff
@@ -572,45 +574,25 @@ const deleteUsers = async (id: string | null) => {
   const result = await usersDeletion.executeMutation(variables);
 
   if (result.error) {
-    const errorMessage =
-      result.error.message === DELETE_USER_FAILED_ONLY_ONE_ADMIN
-        ? t('state.delete_user_failed_only_one_admin')
-        : id
-        ? t('state.delete_user_failure')
-        : t('state.delete_users_failure');
+    const errorMessage = areMultipleUsersSelected.value
+      ? t('state.delete_users_failure')
+      : t('state.delete_user_failure');
     toast.error(errorMessage);
   } else {
     const deletedUsers = result.data?.removeUsersByAdmin || [];
-    const deletedIDs = deletedUsers
+    const deletedUserIDs = deletedUsers
       .filter((user) => user.isDeleted)
       .map((user) => user.userUID);
 
-    const isAdminError = deletedUsers.some(
-      (user) => user.errorMessage === ADMIN_CANNOT_BE_DELETED
-    );
+    handleUserDeletion(deletedUsers, {
+      areMultipleUsersSelected: areMultipleUsersSelected.value,
+      deletedUserIDs,
+    });
 
     usersList.value = usersList.value.filter(
-      (user) => !deletedIDs.includes(user.uid)
+      (user) => !deletedUserIDs.includes(user.uid)
```

Comment:
We could refrain from supplying `deletedUserIDs` via the `metadata` field to the helper function since it is computed based on the `deletedUsers` list; thought to send it along since the computation is already happening at the page level.

Reply:
Here the computation is very cheap, and it won't affect us in any way for computing this inside the function. and if this function is used somewhere else, that user will have to compute this and provide it mostly. which is not ideal imo.

also not having this will help use remove the metadata entirely, and just make the function + types super simple.